src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -398,35 +398,96 @@ private void verifyIsWithinSizeLimitPerEnrollment(List<StudentAttributes> studen
         }
     }
 
-    public void validateSections(List<StudentAttributes> studentList, String courseId) throws EntityDoesNotExistException, EnrollException {
+    /**
+     * Validates both sections and teams for any limit violations
```

Comment:
Perhaps this could be rewritten as `Validates sections for any limit violations and teams for any team name violations`, as `getTeamInvalidiityInfo` checks that there are no same team names accross different sections, rather than if it exceeds a limit.